pylearn2/models/multisubject.py

```diff
@@ -529,9 +530,28 @@ def apply_dropout(self, state, include_prob, scale, theano_rng,
         return T.cast(rval, state.dtype)
 
     @wraps(Layer.cost)
-    def cost(self, Y, Y_hat):
+    def cost(self, Y, Y_hat, other_mlp_weight=1.):
+        cost = None
+        for mlp, ds in safe_izip(model, data):
+            space, sources = self.get_data_specs(mlp)
+            space.validate(ds)
+            (X, Y) = ds
+            Y_hat = mlp.dropout_fprop(
```

Comment:
For this cost, you should be able to use the simpler for-loop from the Default cost rather than from the Dropout one.